.github/workflows/bicep_az_deployment_group.yml

```diff
@@ -0,0 +1,232 @@
+        description: webapp environment variable
+        required: true
+      TFM_UI_URL:
+        description: webapp environment variable
+        required: true
+      AZURE_NUMBER_GENERATOR_FUNCTION_SCHEDULE:
+        description: webapp environment variable
+        required: true
+      SESSION_SECRET:
+        description: webapp environment variable
+        required: true
+      ESTORE_URL:
+        description: webapp environment variable
+        required: true
+
+jobs:
+  configure-command:
+    name: Configure ${{ inputs.environment }} ${{ inputs.action }} command
+    runs-on: [self-hosted, linux, deployment]
+    # TODO: FN-72 NOTE: we don't currently have a `feature` GitHub environment yet so use the `dev` one while we're not doing much
+    # environment: ${{ inputs.environment }}
+    environment: dev
+    outputs:
+      command: ${{ steps.config-command.outputs.command }}
+    steps:
+      - id: config-command
+        run: |
+          if [[ ${{ inputs.action }} == "validate" ]] ; then
+            echo "command=validate" >> $GITHUB_OUTPUT
+          elif [[ ${{ inputs.action }} == "what-if" ]] ; then
+            echo "command=what-if" >> $GITHUB_OUTPUT
+          elif [[ ${{ inputs.action }} == "create" ]] ; then
+            echo "command=create" >> $GITHUB_OUTPUT
+          else
+            echo "::error title=invalid action specified::'${{ inputs.action }}' is not a valid action. Choose from 'validate', 'what-if' or 'create'."
+            exit 1
+          fi
+
+  run-command:
+    name: Run ${{ inputs.environment }} ${{ needs.configure-command.outputs.command }} command
+    needs: configure-command
+    runs-on: [self-hosted, linux, deployment]
+    # TODO: FN-72 NOTE: we don't currently have a `feature` GitHub environment yet so use the `dev` one while we're not doing much
+    # environment: ${{ inputs.environment }}
+    environment: dev
+
+    steps:
+      - uses: actions/checkout@v3
+
+      # We install Bicep ourselves and make sure it works on Alpine linux which is what the runners use.
+      # (See note below for more information.)
+      - name: Install Bicep
+        run: |
+          az version
+          az config set bicep.use_binary_from_path=False
+          az bicep install --target-platform linux-musl-x64
+          az bicep upgrade
+          az bicep version
+
+      - uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Perform deployment ${{ needs.configure-command.outputs.command }} command
+        env:
+          # Note the leading space. We need this to stop Bicep what-if validation failing due to the "["
+          # of the string needing to be escaped if the string is not multi-line (which I believe adds leading spaces as a side effect).
+          # See https://stackoverflow.com/questions/74817321/how-to-escape-leading-square-bracket-in-a-parameter-array-in-bicep
+          # and the issue it mentions for details.
```

Comment:
Aaargh! Again, very confusing, especially when in combination with the GH secret issue on the exact same parameter!